### DISCUSSION TITLE
Deprecation of S3 path-style

### DISCUSSION THREAD

**Discussion by vineetkala11** [aws/aws-sdk-java-v2#3611] [Q&A] [answered]
Recently we migrated from awsSdk version `2.17.284` to `2.18.27` and this migration breaks our `S3AsyncClient.builder`. It because now sdk by default resolving endpoint in `virtual hosted-style`, but as of now our solution expect endpoint in `path-style`. 

Also we found new function `forcePathStyle(Boolean forcePathStyle)` is introduced in `S3BaseClientBuilder`, by providing `true` we explicitly set to `path-style` URL, which solves our issue for now. 👍 

**Now question is**, do we consider `forcePathStyle` as a workaround and in a near future awsSdk will totally deprecate/remove a way to set `path-style` from `S3AsyncClient.builder` or sdk will always come with `forcePathStyle` feature **?** , because we know (as mention in [blog](https://aws.amazon.com/blogs/storage/update-to-amazon-s3-path-deprecation-plan/)) aws recommend to use `virtual hosted-style` and will deprecate `path-style` in future.

Breaking with `2.18.27`

```
S3AsyncClient.builder()
                .endpointOverride(url)
                .credentialsProvider { creds }
                .region(region)
                .httpClient(client)
```

Working with `2.18.27`

```
S3AsyncClient.builder()
                .forcePathStyle(true)
                .endpointOverride(url)
                .credentialsProvider { creds }
                .region(region)
                .httpClient(client)
```

**Comment by debora-ito** [comment, marked as answer]
The `forcePathStyle` will always be an option in the S3 client. We may deprecate it in the future, but it will never be removed. 
What may happen is the S3 service can stop accepting endpoints in path style (like is explained in the [blog post](https://aws.amazon.com/blogs/storage/update-to-amazon-s3-path-deprecation-plan/)), so you'll start seeing service-side exceptions.

**Reply by vineetkala11** [reply]
Thanks @debora-ito , you answered my question 👍 

One last question - is there any update, when (month or year) S3 service stop accepting endpoints in path style, because according to this [blog](https://aws.amazon.com/blogs/storage/update-to-amazon-s3-path-deprecation-plan/) earlier it was planned for September 30, 2020, but I didn't found any final date of deprecation over the internet.

**Reply by debora-ito** [reply]
I don't think there's a timeline yet, I didn't find any announcements after that blog post date.
